### PR TITLE
Update a few python ports which I'm care of

### PR DESCRIPTION
**Pull request by catap**
#### Description

Here an update of a cluster of python ports which I'm care of.

It includes adding `py311` subport for a few non mine ports, which included an update of them also.

I've also fixed building or testing for some of them.

###### Type(s)

- [x] bugfix
- [x] enhancement

###### Tested on

macOS 12.6.3 21G419 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-ipdb/files/ipdb311 [hunk 1]
```diff
@@ -0,0 +1 @@
+bin/ipdb3-3.10
```
Comment (reneeotten) [inline python/py-ipdb/files/ipdb311:1]: this should likely refer to 3.11?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-ipdb/files/ipdb311 (hunk 1 above)
Comment (catap) [reply python/py-ipdb/files/ipdb311:1]: @reneeotten thanks to catch it!

**Comment by catap** [comment]
@reneeotten ping here

**Comment by reneeotten** [comment]
thanks for making the changes and the ping 😉! As usual, we would still wait the 3 days for maintainers of ports you modified here to take a look. But as it has been 4 days now without anyone responding I guess it's time. Thanks for the updates!

**Review by reneeotten — APPROVED** [review]

**Comment by catap** [comment]
@reneeotten thanks! I'll open the second PR in the series of updating my pythong ports soon :)

**Pull request merged**